Enable Google CDN for k8s-artifacts-prod bucket

ameukam opened the pull request:
Related:
  - https://github.com/kubernetes/k8s.io/issues/3327

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

ameukam committed:
Enable Google CDN for k8s-artifacts-prod bucket

Related:
  - https://github.com/kubernetes/k8s.io/issues/3327

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

ameukam commented:
cc @justinsb @johngmyers 
/hold

dims commented:
/approve
/lgtm

endocrimes commented on infra/gcp/bash/ensure-prod-storage-gclb.sh:
```diff
@@ -62,7 +62,7 @@ color 6 "Reconciling GCLB Backend Bucket"
   gcloud --project "${PROD_PROJECT}" compute backend-buckets create "${NAME}" --gcs-bucket-name="${BUCKET_NAME}"
 else
-  gcloud --project "${PROD_PROJECT}" compute backend-buckets update "${NAME}" --gcs-bucket-name="${BUCKET_NAME}"
+  gcloud --project "${PROD_PROJECT}" compute backend-buckets update "${NAME}" --gcs-bucket-name="${BUCKET_NAME}" --enable-cdn --cache-mode=FORCE_CACHE_ALL
```
`FORCE_CACHE_ALL` defaults to a TTL of 1 hour (https://cloud.google.com/cdn/docs/caching), which is probably a fine starting point.

/lgtm

ameukam replied:
@endocrimes targeting a TTL of 30 days for this. https://cloud.google.com/cdn/docs/using-ttl-overrides

ameukam commented:
/hold cancel

ameukam commented:
Deployment done:

```
❯ ./ensure-prod-storage-gclb.sh
Ensuring project exists: k8s-artifacts-prod 
Enabling the compute API: k8s-artifacts-prod 
Reconciling Global Address 
Address: 35.244.192.77
Reconciling GCLB Backend Bucket 
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/backendBuckets/k8s-artifacts-prod].
Reconciling GCLB URL Map 
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/urlMaps/k8s-artifacts-prod].
Reconciling GCLB Target HTTP Proxy 
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/targetHttpProxies/k8s-artifacts-prod].
Reconciling GCLB Forwarding Rule for HTTP 
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/forwardingRules/k8s-artifacts-prod-http].
Reconciling GCLB SSL Certificate 
Reconciling GCLB Target HTTPS Proxy 
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/targetHttpsProxies/k8s-artifacts-prod].
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/targetHttpsProxies/k8s-artifacts-prod].
Reconciling GCLB Forwarding Rule for HTTPS 
Updated [https://www.googleapis.com/compute/v1/projects/k8s-artifacts-prod/global/forwardingRules/k8s-artifacts-prod-https].
Done 
```
